### ISSUE TITLE
Add form page for optional metadata

### ISSUE THREAD

**Issue by nuest** [o2r-project/o2r-UI#146]
For the pilot publication process, it would be most important that the author can edit the DOI before publishing the candidate, see also #130

**Comment by nuest** [comment]
This is solved if #130 works.

**Comment by nuest** [comment]
1. Identify which metadata fields exist, i.e. the content of `metadata` property, list here
1. Decide which ones need to be exposed through a form

**Comment by Fmazin** [comment]
Is [this](https://github.com/o2r-project/o2r-meta/blob/master/schema/docs/elements_o2r.md) a good reference list for the metadata properties?

**Comment by nuest** [comment]
The list you link to is a good start, please update it if you find any information in the API responses that is not documented.

**Comment by Fmazin** [comment]
I have added a optional metadata form page in the create view. It functions like the required metadata. At the moment it contains only the doi field. 
![Bildschirmfoto von 2020-12-11 16-59-35](https://user-images.githubusercontent.com/48436261/101928071-76139600-3bd5-11eb-9695-27b53f7054fa.png)

The doi will also be displayed in the metadata sections of the ERC view.

![Bildschirmfoto von 2020-12-11 17-00-15](https://user-images.githubusercontent.com/48436261/101928360-c68af380-3bd5-11eb-90a3-6aa7b36fe734.png)

This can be expanded to to include the other metadata properties in the list above. But should i keep this general setup?

**Comment by nuest** [comment]
@nuest General setup is alright! I've removed two of the suggested fields (type, metadata license) as these should be set by the system, not the user. (I actually am not sure what "type" is for - is that "workspace" or "compendium"?)

The metadata rendering will include a few links, I assume? Please be aware of https://orcid.org/content/journal-article-display-guidelines and suggest to re-use the DOI logo from the link above the paper, too. Ideally, we can even use https://spdx.org/licenses/ to derive links to the license texts from the license identifier.

**Comment by Fmazin** [comment]
I had some issues with the implementation of the keywords and languages arrays. I tried to do it relatively analog to the [authors](https://github.com/o2r-project/o2r-UI/tree/master/ui/src/components/createERC/requiredMetadata/Authors) in the required metadata section with formik. This unfortunately led to my PC crashing.  Currently consulting with @NJaku01 over the problem.

**Comment by Fmazin** [comment]
Would it be to much if the Languages are checked agianst a list of the [ISO 639-1 ](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) codes?

**Comment by nuest** [comment]
Yes, that validation of user input would be good. Maybe there even is a React module that does this?

If you have to implement this yourself, I suggest a link to the Wikipedia page and then letting the use enter the code, and then create a new issue for a more advanced "user enters first letters of language an then gets suggestions for matching languages and codes".
